UefiPayloadPkg/Library/FdtParserLib/FdtParserLib.c

```diff
@@ -729,6 +744,17 @@ ParsePciRootBridge (
       DEBUG ((DEBUG_INFO, "PciRootBridge->Io.limit %llx, \n", mPciRootBridgeInfo->RootBridge[RbIndex].Io.Limit));
     }
 
+    if (AsciiStrCmp (TempStr, "reg") == 0) {
+      DEBUG ((DEBUG_INFO, "  Found ecam Property TempLen (%08X)\n", TempLen));
+      UINT64  *Data64 = (UINT64 *)(PropertyPtr->Data);
+      mPciRootBridgeInfo->RootBridge[RbIndex].Ecam.Base  = Fdt64ToCpu (*Data64);
+      mPciRootBridgeInfo->RootBridge[RbIndex].Ecam.Limit = Fdt64ToCpu (*(Data64 + 1));
+      mPciRootBridgeInfo->RootBridge[RbIndex].Segment    = RbIndex;

```

Comment:
Agree this is a bug, but multi-rootbridge is not always the same as multi-segment, so we have to compare PciExpressBaseAddress of each rootbridge to confirm if they are same or different segment.

If you already have good idea please update this code, otherwise I would prefer to keep original code (you may override it offline) and I will think about the better implementation so we do not break any existing use cases. (currently most of the UPL use cases are single segment)

Reply:
I could trim this commit out but unless you really think this is breaking something, my suggestion is to keep it the way it is right now which already supports multi segments. In future when we have specific use cases which require update to this logic, they can be added as additional patch. Most of the flows and framework should not alter. WDYT?

Reply:
Actually I suspect this will break single segment use cases as it assigning different segment number to different rootbridge even they are sharing the same segment. Since currently most of the UPL use cases are still single segment, how about we leaving a TODO comment and do not assign segment number for now?

Reply:
I am the first customer with multisegment :). What I am proposing is as follows. For now I have added a check that if ecam is same, do not update segment (I think I need to program segment in case of same ecam- I will udpate in the next patch). This will work as long as we do not have a mix of multi-segment and non-multi segment, which I believe is a safer assumption. And later let us enhance the spec to grab this info from RB node itself. It will be neat. LMK.

Reply:
thanks for taking the feedback and support the code change.